Convert build scripts to Kotlin

lacasseio opened the pull request:
### Context
To familiarize myself with Kotlin DSL and the environment, I'm converting some more build script to Kotlin. Each commit is a project conversion to Kotlin.

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Check&tab=projectOverview&branch_Gradle_Check=lacasseio%2Fkotlin%2Fbuild-script-conversion)
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

bamboo commented on subprojects/launcher/launcher.gradle.kts:
```diff
@@ -0,0 +1,88 @@
+    maxParallelForks = Math.min(3, project.maxParallelForks)
+}
+
+val jar by tasks.existing(Jar::class)
```
:x: Prefer type-safe task accessors: `tasks.jar` is available here.

bamboo commented on subprojects/logging/logging.gradle.kts:
```diff
@@ -0,0 +1,42 @@
+ * Logging infrastructure.
+ */
+plugins {
+    id("java-library")
```
:x: Prefer plugin accessor

bamboo commented on subprojects/model-groovy/model-groovy.gradle.kts:
```diff
@@ -20,26 +20,26 @@ import org.gradle.gradlebuild.unittestandcompile.ModuleType
-    id 'java-library'
-    id 'gradlebuild.strict-compile'
-    id 'gradlebuild.classycle'
+    id("java-library")
```
:x: Prefer plugin accessor

lacasseio commented on subprojects/core/core.gradle.kts:
```diff
@@ -0,0 +1,139 @@
+ * limitations under the License.
+ */
+plugins {
+    id("java-library")
```
Is there a difference between plugin accessor version ``` `java-library` ``` vs `id("java-libary")`?

lacasseio commented:
CI is passing and I addressed all the comment from the @bamboo so I will preemptively merge this to exercise the nightly release tasks and address any further comments in another PR.

bamboo commented:
Looks good, @lacasseio. Additional refinements provided in #7004.
